### PR TITLE
feat: added path and button to open plugin folder in Preferences window

### LINKED ISSUE #3119: Feature Request: Link to Plugin folder in Synfig Preferences
Link to plugin folder in Synfig preferences to help find plugin folder easily for users to install plugins.
Screenshot 2023-07-07 151356

### PR DESCRIPTION
**Pull request by Afterdie**
A path is displayed which is non-editable and a button which opens up a file explorer tab to the path.

closes #3119

### CODE REVIEW

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp [hunk 1]
```diff
@@ -58,6 +58,13 @@
 
 #endif
 
+#ifdef _WIN32
+#include <Windows.h>
+#elif __linux__
+#include <cstdlib>
+#elif __APPLE__
+#include <cstdlib>
```
Comment (rodolforg) [inline synfig-studio/src/gui/dialogs/dialog_setup.cpp:66]: You could use just an `#else`

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp [hunk 2]
```diff
@@ -1446,6 +1467,28 @@ Dialog_Setup::on_brush_path_remove_clicked()
...
+    ShellExecuteA(NULL, "open", "explorer.exe", foldername.c_str(), NULL, SW_SHOWDEFAULT);
+	#elif __linux__
+    std::string command = "xdg-open " + foldername;
+    std::system(command.c_str());
+	#elif __APPLE__
+    std::string command = "open " + foldername;
+    std::system(command.c_str());
+	#else
+	return 0;
+	#endif
```
Comment (rodolforg) [inline synfig-studio/src/gui/dialogs/dialog_setup.cpp:1490]: 1. there is an indentation issue here
2. you used `return 0` in a function that do not return anything (`void`)
3. it does not support OS that do not use UTF-8 encoding :(
4. Try to use `synfig::OS::launch_file_async()` ;)

**Review by rodolforg — CHANGES_REQUESTED** [review]

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp [hunk 3]
```diff
@@ -1446,6 +1461,18 @@ Dialog_Setup::on_brush_path_remove_clicked()
...
+void
+Dialog_Setup::on_plugin_path_change_clicked()
+{
+	String foldername = ResourceHelper::get_plugin_path();
+	//processes the path from get_plugin_path()
+	size_t pos = foldername.find('/');
+    while (pos != std::string::npos) {
+        foldername.replace(pos, 1, "\\");
+        pos = foldername.find('/', pos + 2);
+    }
```
Comment (rodolforg) [inline synfig-studio/src/gui/dialogs/dialog_setup.cpp:1473]: Is this really needed on Windows? (It won't work on Linux or MacOS by the way)

Besides, there are indentation issues.

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp [hunk 4]
```diff
@@ -58,6 +58,7 @@
 
 #endif
 
+#include <synfig/os.h>
```
Comment (rodolforg) [inline synfig-studio/src/gui/dialogs/dialog_setup.cpp:61]: Move this line up, right above this line : `#include <synfig/rendering/renderer.h>`

We try to group #includes and sort them alphabetically.

**Review by rodolforg — CHANGES_REQUESTED** [review]

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp (hunk 3 above)
Comment (rodolforg) [reply synfig-studio/src/gui/dialogs/dialog_setup.cpp:1473]: So these slash-to-backslash replacement is really needed, right?

Comment (Afterdie) [reply synfig-studio/src/gui/dialogs/dialog_setup.cpp:1473]: no this isnt needed i tried on windows and it works without it. Im asking if theres some way to add it for mac linux

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp (hunk 4 above)
Comment (Afterdie) [reply synfig-studio/src/gui/dialogs/dialog_setup.cpp]: @rodolforg the master branch had some imports which werent in alphabetical order and also it was causing conflicts with my pr so i put in os.h in the empty line to resolve it

**Review by rodolforg — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**